packages/common/src/location/hash_location_strategy.ts

```diff
@@ -43,9 +45,14 @@ export class HashLocationStrategy extends LocationStrategy {
     }
   }
 
+  ngOnDestroy(): void {
+    this._removeListenerFns.forEach(fn => fn());
+    this._removeListenerFns = [];
```

Comment:
```suggestion
    while (this._removeListenerFns.length) {
      this._removeListenerFns.pop()();
    }
```

I think the above is better because if `fn` throws exception than this is left in inconsistent state. Where as with the above change it is clear what to do next.